cfg/config.go

```diff
@@ -2,45 +2,200 @@ package cfg
+	if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+
+	// At least one file must exist
+	if remotesBytes == nil && projectBytes == nil {
+		return nil, errors.New(NoInertiaConfig)
+	}
+
+	var (
+		project = &common.InertiaProject{}
+		remotes = &InertiaRemotes{}
+	)
+
+	// If only project is present, construct config using project
+	if projectBytes != nil && remotesBytes == nil {
+		err = toml.Unmarshal(projectBytes, project)
+		if err != nil {
+			return nil, fmt.Errorf("project config error: %s", err.Error())
+		}
+		return NewConfigFromTOML(*project, InertiaRemotes{Version: project.Version})
+	}
+
+	// If only remotes is present, construct config using remotes
+	if projectBytes == nil && remotesBytes != nil {
+		err = toml.Unmarshal(remotesBytes, remotes)
+		if err != nil {
+			return nil, fmt.Errorf("remotes config error: %s", err.Error())
+		}
+		return NewConfigFromTOML(common.InertiaProject{
+			Version: remotes.Version,
+		}, *remotes)
+	}
+
+	// If both files are present, construct config using both
+	err = toml.Unmarshal(projectBytes, project)
+	if err != nil {
+		return nil, fmt.Errorf("project config error: %s", err.Error())
+	}
+	err = toml.Unmarshal(remotesBytes, remotes)
+	if err != nil {
+		return nil, fmt.Errorf("remotes config error: %s", err.Error())
+	}
+	return NewConfigFromTOML(*project, *remotes)
+}
+
+// NewConfigFromTOML loads configuration from TOML format structs
+func NewConfigFromTOML(project common.InertiaProject, remotes InertiaRemotes) (*Config, error) {
+	// Set remote defaults
+	if remotes.Remotes == nil {
+		r := make(map[string]*RemoteVPS)
+		remotes.Remotes = r
+	}
+	if remotes.Version == nil {
+		remotes.Version = project.Version
+	}
+
+	// Check all is g
+	if *project.Version != *remotes.Version {
+		return nil, fmt.Errorf("mismatching versions %s and %s", *project.Version, *remotes.Version)
+	}
+
+	// Generate configuration
+	return &Config{
+		Version:       common.Dereference(project.Version),
+		Project:       common.Dereference(project.Project),
+		BuildType:     common.Dereference(project.Build.Type),
+		BuildFilePath: common.Dereference(project.Build.ConfigPath),
+		RemoteURL:     common.Dereference(project.Repository.RemoteURL),
+		remotes:       remotes.Remotes,
+	}, nil
+}
+
+// GetRemotes retrieves a list of all remotes
+func (config *Config) GetRemotes() []*RemoteVPS {
+	remotes := make([]*RemoteVPS, 0)
+	for name, remote := range config.remotes {
+		// Set name
+		remote.Name = name
+		remotes = append(remotes, remote)
+	}
+	return remotes
+}
+
+// GetRemote retrieves a remote by name
+func (config *Config) GetRemote(name string) (*RemoteVPS, bool) {
+	for n, remote := range config.remotes {
+		if n == name {
+			// Set name
+			remote.Name = n
+			return remote, true
+		}
+	}
+	return nil, false
+}
+
+// AddRemote adds a remote to configuration
+func (config *Config) AddRemote(remote *RemoteVPS) bool {
+	_, ok := config.remotes[remote.Name]
+	if ok {
```

Comment:
Is this `ok` supposed to connote that the remote exists? Why not just call it `exists` or `isset` etc?

Reply:
I think it's just a standard, though perhaps `found` would be more appropriate